### ISSUE TITLE
Reg: Backprop in generator network

### ISSUE THREAD

**Issue by vasavig** [reedscot/icml2016#8]
Hi,

I have a question regarding the differential which is propagated back in the generator network. The first element of df_dg table, df_dg[1] is passed to the backward function of generator. Is this the differential of loss function with respect to the input image to discriminator (which is the first input to the discriminator)? If so, df_dg[2] would be the differential of the loss function with respect to the second input to discriminator, the text attribute vector, right?

Thanks!
